ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java

```diff
@@ -673,6 +674,9 @@ public void invokeCallback(Exception exception) {
         futureResult.done(totalBytesRead.get(), exception);
         if (callback != null) {
           callback.onCompletion(totalBytesRead.get(), exception);
+          // the callback may hold a reference to a large buffer. Setting the callback to null allows for memory to be
+          // freed before the next request comes over this connection.
+          callback = null;
```

Comment:
I wonder if we can add junit tests for this by using a `WeakReference` to see if an object has been `enqueued` or not....

Reply:
I think it might be hard to make such a test deterministic, but I will keep thinking about if there is a way to better test this. There is always reflection, but that is a little intrusive into the inner structure of the class being tested.